src/ducks/transactions/TransactionsPage.jsx

```diff
@@ -109,6 +141,62 @@ class TransactionsPage extends Component {
       .filter(Boolean)
   }
 
+  handleChangeTopmostTransaction = transaction => {
+    this.setState({
+      currentMonth: getMonth(transaction.date)
+    })
+  }
+
+  handleIncreaseLimitMax = () => {
+    this.setState({
+      limitMax: this.state.limitMax + 10
+    })
+  }
+
+  handleDecreaseLimitMin = () => {
+    this.setState({
+      limitMin: Math.max(this.state.limitMin - 5, 0)
+    })
+  }
+
+  handleChangeMonth = month => {
+    const transactions = this.props.filteredTransactions
+    const findMonthIndex = month =>
+      findIndex(transactions, t => t.date.indexOf(month) === 0)
+    let limitMin = findMonthIndex(month)
+
+    if (limitMin == -1) {
+      const validMonths = uniq(transactions.map(x => getMonth(x.date)))
+      const filterBefore = x => x < month
+      const filterAfter = x => x > month
+      const filters =
+        month < this.state.currentMonth
+          ? [filterBefore, filterAfter]
+          : [filterAfter, filterBefore]
+      for (let filter of filters) {
+        const months = validMonths.filter(filter)
+        if (months.length > 0) {
+          month = months[0]
+          break
+        }
+      }
+      limitMin = findMonthIndex(month)
+    }
```

Comment:
Can you explain what's going on here ? Actually I understand that you are trying to get something if the requested month has no transactions, but I don't understand the filtering part